language/tools/move-package/src/resolution/resolution_graph.rs

```diff
@@ -769,3 +770,22 @@ impl ResolvedPackage {
         }
     }
 }
+
+impl ResolvedGraph {
+    pub fn file_sources(&self) -> BTreeMap<FileHash, (Symbol, String)> {
+        self.package_table
+            .iter()
+            .flat_map(|(_, rpkg)| {
+                rpkg.get_sources(&self.build_options)
+                    .unwrap()
+                    .iter()
+                    .map(|fname| {
+                        let contents = read_to_string(Path::new(fname.as_str())).unwrap();
+                        let fhash = FileHash::new(&contents);
+                        (fhash, (*fname, contents))
+                    })
+                    .collect::<BTreeMap<_, _>>()
```

Comment:
Do you need this inner collect? Just curious, what error do you get without it?

Reply:
I am pretty sure this was inspired by other code pieces in the same file, rather than a result of a deep thought :-) There is an error, though, if collect is not there:
```
cannot return value referencing temporary value
```

In fact, adding the collect call is what is suggested by flycheck